### DISCUSSION TITLE
How to use differentiable simulation?

### DISCUSSION THREAD

**Discussion by HerocatUED** [Genesis-Embodied-AI/genesis-world#691] [Q&A]
After we set ```requires_grad=True``` in ```SimOptions```, how can we get gradient from gs scene or env?
Say, we have ```action_t = model.act(state_t)``` and ```state_{t+1} = env.step(action_t)```, our loss function is L2 norm between state_{t+1} and target_{t+1}, how can we calculate gradient to optimize our model or policy?

**Comment by HerocatUED** [comment]
Oh, I forgot to point out that the env above is for a unitree-Go2 locomotion task. Additionally, I don't know which method should I use for rigid body simulation. I just use default settings as example does. When I check API doc, I noticed that some solver is not differentiable currently. And I know nothing about physics solvers. So, I wonder will there be any examples for differentiable simulation.

**Comment by Cfomodz** [comment]


**Reply by duburcqa** [reply]
This answer is wrong. Rigid body simulation is now fully differentiable. Please have to look at [tests/test_grad.py](https://github.com/Genesis-Embodied-AI/Genesis/blob/main/tests/test_grad.py) as reference. Official documentation will come soon.
